internal/policygen/iam.go

```diff
@@ -95,54 +95,120 @@ func generateIAMPolicies(rn runner.Runner, resources []*states.Resource, outputP
 
 	// All roles associated with a root resource (organization, folder or project).
-	var bindings = make(map[root]roleBindings)
+	var allBindings = make(map[root]roleBindings)
 
 	typeToIDField := map[string]string{
 		"project":      "project",
 		"folder":       "folder",
 		"organization": "org_id",
 	}
 
-	for t, idField := range typeToIDField {
-		// TODO(https://github.com/GoogleCloudPlatform/healthcare-data-protection-suite/issues/152): Support google_*_iam_bindings.
-		resourceType := fmt.Sprintf("google_%s_iam_member", t)
-		instances, err := terraform.GetInstancesForType(resources, resourceType)
+	for rootType, idField := range typeToIDField {
+		IAMMembers, err := members(rn, resources, rootType, idField)
 		if err != nil {
-			return nil, fmt.Errorf("get resource instances for type %q: %v", resourceType, err)
+			return nil, err
 		}
 
-		for _, ins := range instances {
-			if err := validate(ins, []string{idField, "role", "member"}); err != nil {
-				return nil, err
-			}
+		IAMBindings, err := bindings(rn, resources, rootType, idField)
+		if err != nil {
+			return nil, err
+		}
 
-			id := ins[idField].(string) // Type checked in validate()
-			// For projects, the ID in the state is the project ID, but we need project number in policies.
-			if t == "project" {
-				if id, err = projectNumber(rn, id); err != nil {
-					return nil, err
+		// Add IAMBindings to IAMMembers.
+		// If IAMMembers have members for the same root and role, replace it with the value from IAMBindings.
+		for root, bindings := range IAMBindings {
+			for role, members := range bindings {
+				// Init the roleBindings map if it didn't exist.
+				if _, ok := IAMMembers[root]; !ok {
+					IAMMembers[root] = make(roleBindings)
 				}
-			} else if t == "folder" {
-				// For folders, the ID can be either {folder_id} or folders/{folder_id}. Remove the 'folders/' prefix if exists.
-				id = strings.TrimPrefix(id, "folders/")
+				IAMMembers[root][role] = members
 			}
+		}
+		for root, bindings := range IAMMembers {
+			allBindings[root] = bindings
+		}
+	}
+	return allBindings, nil
+}
 
-			key := root{Type: t, ID: id}
+// members returns role bindings map for google_%s_iam_member (non-authoritative).
+func members(rn runner.Runner, resources []*states.Resource, rootType, idField string) (map[root]roleBindings, error) {
+	var bindings = make(map[root]roleBindings)
+	resourceType := fmt.Sprintf("google_%s_iam_member", rootType) // non-authoritative
+	instances, err := terraform.GetInstancesForType(resources, resourceType)
+	if err != nil {
+		return nil, fmt.Errorf("get resource instances for type %q: %v", resourceType, err)
+	}
 
-			// Init the roleBindings map if it didn't exist.
-			if _, ok := bindings[key]; !ok {
-				bindings[key] = make(roleBindings)
-			}
+	for _, ins := range instances {
+		if err := validateMandatoryStringFields(ins, []string{idField, "role", "member"}); err != nil {
+			return nil, err
```

Comment:
Maybe add "fmt.Errorf()" to indicate this error comes from missing fields in the instance.

Reply:
It's already indicated in the `validateMandatoryStringFields` function that it's a mandatory field missing or wrong value type. It's a bit verbose to repeat the same thing here.